File: mobile/androidApp/src/main/java/com/vanessa/ui/screens/planning/PlanningScreen.kt
package com.vanessa.ui.screens.planning

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import com.vanessa.core.ui.ThemeTokens
import com.vanessa.core.ui.formatBRL
import com.vanessa.di.AppContainer

@Composable
fun PlanningScreen(container: AppContainer) {
    val state by container.planningVm.state.collectAsStateWithLifecycle()
    val scope = rememberCoroutineScope()
    var addingPiggy by remember { mutableStateOf(false) }
    var addingGoal by remember { mutableStateOf(false) }
    var addingFixed by remember { mutableStateOf(false) }

    LazyColumn(
        Modifier
            .fillMaxSize()
            .background(Color(ThemeTokens.palette.background))
            .padding(16.dp)
    ) {
        item {
            Text("Planejamento",
                color = Color(ThemeTokens.palette.textPrimary),
                style = MaterialTheme.typography.headlineSmall,
                fontWeight = FontWeight.SemiBold)
            Spacer(Modifier.height(8.dp))
            Surface(color = Color(ThemeTokens.palette.surface),
                shape = RoundedCornerShape(12.dp),
                modifier = Modifier.fillMaxWidth()) {
                Column(Modifier.padding(16.dp)) {
                    Text("Receita do mês",
                        color = Color(ThemeTokens.palette.textSecondary))
                    Text(formatBRL(state.monthlyIncome),
                        color = Color(ThemeTokens.palette.success),
                        style = MaterialTheme.typography.headlineMedium,
                        fontWeight = FontWeight.Bold)
                    if (state.reminders.overdue.isNotEmpty()) {
                        Text("${state.reminders.overdue.size} vencidos",
                            color = Color(ThemeTokens.palette.danger),
                            fontWeight = FontWeight.SemiBold)
                    }
                    if (state.reminders.dueToday.isNotEmpty()) {
                        Text("${state.reminders.dueToday.size} vencendo hoje",
                            color = Color(ThemeTokens.palette.warning))
                    }
                }
            }
        }
        item {
            Spacer(Modifier.height(20.dp))
            Row(verticalAlignment = Alignment.CenterVertically) {
                Text("Cofrinhos", color = Color(ThemeTokens.palette.textSecondary))
                Spacer(Modifier.weight(1f))
                OutlinedButton(onClick = { addingPiggy = true }) { Text("+ Novo") }
            }
        }
        items(state.piggy, key = { it.id }) { pb ->
            Surface(
                color = Color(ThemeTokens.palette.surface),
                shape = RoundedCornerShape(12.dp),
                modifier = Modifier.fillMaxWidth().padding(vertical = 6.dp)
            ) {
                Column(Modifier.padding(16.dp)) {
                    Text(pb.name, color = Color(ThemeTokens.palette.textPrimary),
                        fontWeight = FontWeight.SemiBold)
                    Text("${formatBRL(pb.savedAmount)} de ${formatBRL(pb.target)}",
                        color = Color(ThemeTokens.palette.textSecondary))
                    Spacer(Modifier.height(8.dp))
                    val pct = if (pb.targetAmount > 0) (pb.savedAmount / pb.targetAmount).coerceIn(0.0, 1.0) else 0.0
                    Box(
                        Modifier
                            .fillMaxWidth()
                            .height(8.dp)
                            .clip(RoundedCornerShape(4.dp))
                            .background(Color(ThemeTokens.palette.surfaceElevated))
                    ) {
                        Box(
                            Modifier
                                .fillMaxHeight()
                                .fillMaxWidth(pct.toFloat())
                                .background(Color(ThemeTokens.palette.lavender))
                        )
                    }
                    Spacer(Modifier.height(8.dp))
                    Row {
                        OutlinedButton(onClick = {
                            container.planningVm.deposit(pb.id, 50.0)
                        }) { Text("+R$50") }
                        Spacer(Modifier.width(8.dp))
                        TextButton(onClick = { container.planningVm.removePiggy(pb.id) }) {
                            Text("Excluir", color = Color(ThemeTokens.palette.danger))
                        }
                    }
                }
            }
        }
        item {
            Spacer(Modifier.height(20.dp))
            Row(verticalAlignment = Alignment.CenterVertically) {
                Text("Metas", color = Color(ThemeTokens.palette.textSecondary))
                Spacer(Modifier.weight(1f))
                OutlinedButton(onClick = { addingGoal = true }) { Text("+ Nova") }
            }
        }
        items(state.goals, key = { it.id }) { gl ->
            Surface(
                color = Color(ThemeTokens.palette.surface),
                shape = RoundedCornerShape(12.dp),
                modifier = Modifier.fillMaxWidth().padding(vertical = 6.dp)
            ) {
                Row(Modifier.padding(16.dp), verticalAlignment = Alignment.CenterVertically) {
                    Column(Modifier.weight(1f)) {
                        Text(gl.title, color = Color(ThemeTokens.palette.textPrimary))
                        Text("${formatBRL(gl.targetAmount)} em ${gl.targetMonths} meses",
                            color = Color(ThemeTokens.palette.textSecondary))
                    }
                    IconButton(onClick = { container.planningVm.removeGoal(gl.id) }) {
                        Text("🗑", color = Color(ThemeTokens.palette.danger))
                    }
                }
            }
        }
        item {
            Spacer(Modifier.height(20.dp))
            Row(verticalAlignment = Alignment.CenterVertically) {
                Text("Gastos fixos", color = Color(ThemeTokens.palette.textSecondary))
                Spacer(Modifier.weight(1f))
                OutlinedButton(onClick = { addingFixed = true }) { Text("+ Novo") }
            }
        }
        items(state.fixed, key = { it.id }) { fc ->
            Surface(
                color = Color(ThemeTokens.palette.surface),
                shape = RoundedCornerShape(12.dp),
                modifier = Modifier.fillMaxWidth().padding(vertical = 6.dp)
            ) {
                Row(Modifier.padding(16.dp), verticalAlignment = Alignment.CenterVertically) {
                    Column(Modifier.weight(1f)) {
                        Text(fc.name, color = Color(ThemeTokens.palette.textPrimary))
                        Text("${formatBRL(fc.amount)} dia ${fc.dueDay}",
                            color = Color(ThemeTokens.palette.textSecondary))
                    }
                    TextButton(onClick = { container.planningVm.markFixedPaid(fc.id) }) {
                        Text("Pagar")
                    }
                    IconButton(onClick = { container.planningVm.removeFixed(fc.id) }) {
                        Text("🗑", color = Color(ThemeTokens.palette.danger))
                    }
                }
            }
        }
        item { Spacer(Modifier.height(80.dp)) }
    }

    if (addingPiggy) TextInputDialog("Novo cofrinho", { name -> }) { addingPiggy = false }
    if (addingGoal) TextInputDialog("Nova meta", { name -> }) { addingGoal = false }
    if (addingFixed) TextInputDialog("Novo gasto fixo", { name -> }) { addingFixed = false }
}

@Composable
private fun TextInputDialog(title: String, onConfirm: (String) -> Unit, onDismiss: () -> Unit) {
    var text by remember { mutableStateOf("") }
    AlertDialog(
        onDismissRequest = onDismiss,
        title = { Text(title) },
        text = {
            OutlinedTextField(value = text, onValueChange = { text = it },
                label = { Text("Nome") },
                singleLine = true, modifier = Modifier.fillMaxWidth())
        },
        confirmButton = { Button(onClick = { onConfirm(text); onDismiss() }) { Text("Salvar") } },
        dismissButton = { TextButton(onClick = onDismiss) { Text("Cancelar") } }
    )
}

File: mobile/iosApp/Vanessa/Screens/PlanningScreen.swift
import Foundation
import SwiftUI
import VanessaShared
import shared

struct PlanningScreen: View {
    @StateObject private var vm: KmpViewModel<PlanningUiState>
    let kmm: PlanningViewModel
    let scope = AppRuntime.shared_kmp
    @State private var addingPiggy = false
    @State private var addingGoal = false
    @State private var addingFixed = false

    init(container: IosContainer) {
        self.kmm = container.planningVm
        _vm = StateObject(wrappedValue: KmpViewModel(initial: container.planningVm.snapshot(),
                                                     poll: { container.planningVm.snapshot() }))
    }

    var body: some View {
        ScrollView {
            VStack(alignment: .leading, spacing: 20) {
                Text("Planejamento").font(.title3).bold().foregroundColor(ThemePalette.textPrimary)
                summary
                piggies
                goals
                fixed
                Spacer(minLength: 40)
            }.padding(16)
        }
        .sheet(isPresented: $addingPiggy) {
            TextInputSheet(title: "Novo cofrinho", onConfirm: { name in
                kmm.addPiggyBank(name: name ?? "Cofrinho", saved: 0.0, target: 100.0)
                addingPiggy = false
            })
        }
        .sheet(isPresented: $addingGoal) {
            TextInputSheet(title: "Nova meta", onConfirm: { name in
                kmm.addGoal(title: name ?? "Meta", type: "viagem", target: 1000.0, targetMonths: 12)
                addingGoal = false
            })
        }
        .sheet(isPresented: $addingFixed) {
            TextInputSheet(title: "Novo gasto fixo", onConfirm: { name in
                kmm.addFixed(name: name ?? "Gasto", amount: 100.0, dueDay: 5, category: "outros")
                addingFixed = false
            })
        }
    }

    private var summary: some View {
        VStack(alignment: .leading) {
            Text("Receita do mês").foregroundColor(ThemePalette.textSecondary)
            Text(formatBRL(vm.current.monthlyIncome))
                .font(.system(size: 28, weight: .bold))
                .foregroundColor(ThemePalette.success)
            if !vm.current.reminders.overdue.isEmpty {
                Text("\(vm.current.reminders.overdue.count) vencidos")
                    .foregroundColor(ThemePalette.danger)
            }
            if !vm.current.reminders.dueToday.isEmpty {
                Text("\(vm.current.reminders.dueToday.count) vencendo hoje")
                    .foregroundColor(ThemePalette.warning)
            }
        }.padding(20).frame(maxWidth: .infinity, alignment: .leading).background(ThemePalette.surface.cornerRadius(12))
    }

    private var piggies: some View {
        VStack(alignment: .leading) {
            HStack {
                Text("Cofrinhos").foregroundColor(ThemePalette.textSecondary)
                Spacer()
                Button("+ Novo") { addingPiggy = true }
            }
            ForEach(vm.current.piggy, id: \.id) { pb in
                VStack(alignment: .leading) {
                    Text(pb.name).foregroundColor(ThemePalette.textPrimary).bold()
                    Text("\(formatBRL(pb.savedAmount)) de \(formatBRL(pb.targetAmount))")
                        .foregroundColor(ThemePalette.textSecondary)
                    ProgressBar(ratio: pb.targetAmount > 0
                                ? CGFloat(pb.savedAmount / pb.targetAmount) : 0)
                    HStack {
                        Button("+R$50") { kmm.deposit(id: pb.id, amount: 50.0) }
                        Spacer()
                        Button("Excluir") { kmm.removePiggy(id: pb.id) }
                    }.font(.caption)
                }.padding().background(ThemePalette.surface.cornerRadius(12))
            }
        }
    }

    private var goals: some View {
        VStack(alignment: .leading) {
            HStack {
                Text("Metas").foregroundColor(ThemePalette.textSecondary)
                Spacer()
                Button("+ Nova") { addingGoal = true }
            }
            ForEach(vm.current.goals, id: \.id) { g in
                HStack {
                    VStack(alignment: .leading) {
                        Text(g.title).foregroundColor(ThemePalette.textPrimary)
                        Text("\(formatBRL(g.targetAmount)) em \(g.targetMonths) meses")
                            .foregroundColor(ThemePalette.textSecondary).font(.caption)
                    }
                    Spacer()
                    Button(action: { kmm.removeGoal(id: g.id) }) { Text("🗑") }
                }.padding().background(ThemePalette.surface.cornerRadius(12))
            }
        }
    }

    private var fixed: some View {
        VStack(alignment: .leading) {
            HStack {
                Text("Gastos fixos").foregroundColor(ThemePalette.textSecondary)
                Spacer()
                Button("+ Novo") { addingFixed = true }
            }
            ForEach(vm.current.fixed, id: \.id) { f in
                HStack {
                    VStack(alignment: .leading) {
                        Text(f.name).foregroundColor(ThemePalette.textPrimary)
                        Text("\(formatBRL(f.amount)) dia \(f.dueDay)")
                            .foregroundColor(ThemePalette.textSecondary).font(.caption)
                    }
                    Spacer()
                    Button("Pagar") { kmm.markFixedPaid(id: f.id) }
                    Button(action: { kmm.removeFixed(id: f.id) }) { Text("🗑") }
                }.padding().background(ThemePalette.surface.cornerRadius(12))
            }
        }
    }
}

struct ProgressBar: View {
    let ratio: CGFloat
    var body: some View {
        GeometryReader { geo in
            ZStack(alignment: .leading) {
                RoundedRectangle(cornerRadius: 4).fill(ThemePalette.surfaceElevated)
                RoundedRectangle(cornerRadius: 4).fill(ThemePalette.lavender)
                    .frame(width: min(geo.size.width * ratio, geo.size.width))
            }
        }.frame(height: 8)
    }
}

struct TextInputSheet: View {
    let title: String
    let onConfirm: (String?) -> Void
    @State private var text = ""
    var body: some View {
        NavigationView {
            Form {
                TextField("Nome", text: $text)
            }
            .navigationTitle(title)
            .toolbar {
                ToolbarItem(placement: .confirmationAction) {
                    Button("Salvar") { onConfirm(text.isEmpty ? nil : text) }
                }
            }
        }
    }
}
